### PR TITLE
Join the shared room on each node create -- build based room name and key

### PR DESCRIPTION
**Pull request by andrewxhill**

### CODE REVIEW

**Comment by andrewxhill** [comment]
okay, this one is ready too

File: App/Sagas/TextileSagas.js [hunk 1]
```diff
@@ -124,6 +125,7 @@ export function * createNode ({path}) {
     const debugLevel = (__DEV__ ? "DEBUG" : "INFO")
     const createNodeSuccess = yield call(IPFS.createNodeWithDataDir, path, API_URL, debugLevel)
     if (createNodeSuccess) {
+      yield call(IPFS.updateThread, Config.ALL_THREAD_MNEMONIC, Config.ALL_THREAD_NAME)
```
Comment (asutula) [inline App/Sagas/TextileSagas.js]: Should we be checking the return value here? If it fails, it should run the `else` below. So just to be safe, something like: move the call to `updateThread` out of this `if` and then update the `if` to `if (createNodeSuccess && updateThreadSuccess)`

File: ios/TextilePhotos.xcodeproj/xcshareddata/xcschemes/TextilePhotos.xcscheme [hunk 2]
```diff
@@ -81,7 +90,7 @@
       </AdditionalOptions>
    </TestAction>
    <LaunchAction
-      buildConfiguration = "Debug"
+      buildConfiguration = "Release"
```
Comment (asutula) [inline ios/TextilePhotos.xcodeproj/xcshareddata/xcschemes/TextilePhotos.xcscheme]: Did you change the scheme's 'run' action to `Release` and forget to change it back?

Comment (andrewxhill) [reply ios/TextilePhotos.xcodeproj/xcshareddata/xcschemes/TextilePhotos.xcscheme]: so there are now two schemes. for the 'release' scheme, i guess you'd still want a debug build for logs etc?

Comment (asutula) [reply ios/TextilePhotos.xcodeproj/xcshareddata/xcschemes/TextilePhotos.xcscheme]: Oh, right. Let me take a step back here and make sure I understand. Why do we need a new scheme? Within a single scheme, can't we load a different set of env vars based on whether it's a debug or release build? Maybe I don't understand the way react-native-config works...

Comment (asutula) [reply ios/TextilePhotos.xcodeproj/xcshareddata/xcschemes/TextilePhotos.xcscheme]: I'm guessing that doesn't achieve some goal you had in mind, just want to understand what that is.

File: App/Sagas/TextileSagas.js (hunk 1 above)
Comment (andrewxhill) [reply App/Sagas/TextileSagas.js]: cool, fixed

File: ios/TextilePhotos.xcodeproj/xcshareddata/xcschemes/TextilePhotos.xcscheme (hunk 2 above)
Comment (asutula) [reply ios/TextilePhotos.xcodeproj/xcshareddata/xcschemes/TextilePhotos.xcscheme]: I guess the way I'd think about the different schemes is not as 'release' and 'debug', but different environments, 'dev' and 'production'. You might want to run either scheme in debug or release mode depending on what you're doing. For that reason, I'd leave the 'run' action set to 'debug' so things behave as usual with Xcode development.

Comment (andrewxhill) [reply ios/TextilePhotos.xcodeproj/xcshareddata/xcschemes/TextilePhotos.xcscheme]: cool, updated

**Pull request merged**